Update readme (#81)
Nope-Cardgame/Backend: README.md

Einen [beispielhaften Ablauf](https://github.com/Nope-Cardgame/Doku/blob/main/Schnittstellen/Sequenzdiagramm.md) der Kommunikation zwischen diesem Server und den Clients ist ebenfalls dokumentiert.

## Aufgabenverteilung

> An dem Backend haben hauptsächlich die Entwickler Hannes Nowak, Frederik Hartmann und Fabian Winkelmann gearbeitet. Es ist nicht möglich zu differenzieren welches Teammitglied an welchem Feature gearbeitet, da die Arbeit hauptsächlich im Team stattgefunden hat. Oft wurde Pair Programmierung betrieben, sodass die Arbeit nicht mehr zuzuordnen ist. Die Pullrequest wurden von allen Teammitgliedern geprüft, debugged und getestet.  

Reviewer: ```suggestion
An dem Backend haben hauptsächlich die Entwickler Hannes Nowak, Frederik Hartmann und Fabian Winkelmann gearbeitet. Es ist nicht möglich zu differenzieren welches Teammitglied an welchem Feature gearbeitet, da die Arbeit hauptsächlich im Team stattgefunden hat. Oft wurde Pair Programmierung betrieben, sodass die Arbeit nicht mehr zuzuordnen ist. Pull Requests wurden von allen Teammitgliedern geprüft, debugged und getestet.  
```

> Im folgenden haben wir versucht zu unterteilen, wer hauptsächlich an welchem Feature gearbeitet hat. Uns ist dies schwer gefallen, da wir oft gemeinsam an einer Feature gearbeitet haben und jeder in Form von Ideen und Vorschlägen an jedem Feature beteiligt war. Oft hat ein Entwickler angefangen ein Feature zu implementieren und ein anderer hat es dann fertiggestellt, debugged oder getestet. Die folgende Auflisten zeigt, wer hauptsächlich an welchem Feature gearbeitet hat.

Reviewer: ```suggestion
Im folgenden haben wir versucht zu unterteilen, wer hauptsächlich an welchem Feature gearbeitet hat. Uns ist dies schwer gefallen, da wir oft gemeinsam an einem Feature gearbeitet haben und jeder in Form von Ideen und Vorschlägen an jedem Feature beteiligt war. Oft hat ein Entwickler angefangen ein Feature zu implementieren und ein anderer hat es dann fertiggestellt, debugged oder getestet. Die folgende Auflistung versucht zu zeigen, wer hauptsächlich an welchem Feature gearbeitet hat.
```

| Aufgabe                                           | Hauptentwickler            |
| ------------------------------------------------- | -------------------------- |
| RepoTemplate                                      | Hannes                     |
| Datenbankklassen                                  | Hannes                     |
| Struktur des Templates                            | Fabian                     |
| Initialisieren von Socket.io                      | Fabian                     |
| Anmelden, Registrieren                            | Fabian                     |
| ER-Diagramm                                       | Hannes, Frederik           |
| Sequenzdiagramm                                   | Frederik                   |
| Schnittstellen                                    | Frederik,   Hannes, Fabian |
| Aktivitätsdiagramm,  Use Case                     | Fabian                     |
| State-Charts                                      | Frederik                   |
| Zod                                               | Hannes                     |
| XState                                            | Hannes                     |

> | Freddy                                            | Gamelogic                  |

Reviewer: ```suggestion
| Gamelogic                                         | Frederik                   |
```
